### DISCUSSION TITLE
Replace Prefix Table for Query in common CMS

### DISCUSSION THREAD

**Discussion by valentino-90** [mtxr/vscode-sqltools#952] [Ideas]
Run query from code, with a prompt input for pre-replace table prefix (joomla #__, wordpress => {$wpdb->prefix})  before run

**Comment by ppcamp** [comment]
A response a bit outdated, but I suppose that there's a tool for this already, change your settings to something like this in your workspace

```bash
    "sqltools.queryParams.enableReplace": true,
    "sqltools.queryParams.regex": "\\$[\\d]+|([@:][\\w]+)|(\\$\\[[\\w]+\\])",
    "sqltools.connections": [
        {
            "previewLimit": 50,
            "server": "localhost",
            "port": 5432,
            "driver": "PostgreSQL",
            "name": "MyDb",
            "group": "Locals",
            "database": "world-db",
            "username": "world",
            "password": "world123",
            "askForPassword": false,
            "variables": {
                ":name": "'%e%'",
            }
        }
    ]
```

Then, if I do the following query, it'll automatically match and replace the ":name" param, but will ask you what to replace, in this case, I need to put the column name

```sql
SELECT *
FROM city
WHERE $[whatToSearch] ILIKE :name;
```

See

https://github.com/mtxr/vscode-sqltools/blob/dev/packages/plugins/connection-manager/extension.ts#L221

https://github.com/mtxr/vscode-sqltools/blob/dev/packages/util/query/index.ts#L58
